### PR TITLE
Fix cleaning not initialized http2 context.

### PR DESCRIPTION
**Pull request by EvgeniiMekhanik**
We initialize http2 context in `tfw_tls_over`, but
if this function was not called, because tls handshake
fails we clean uninitialized context, that can lead to
kernel BUG.

### CODE REVIEW

**Comment by EvgeniiMekhanik** [comment]
Also we can revert moving `tfw_h2_context_init` from last commit in master instead of this patch

**Comment by EvgeniiMekhanik** [comment]
We also can move `tfw_h2_context_init` back to `tfw_tls_conn_init`

**Review by const-t — APPROVED** [review]

File: fw/tls.c [hunk 1]
```diff
@@ -611,9 +611,11 @@ tfw_tls_conn_dtor(void *c)
 {
 	struct sk_buff *skb;
 	TlsCtx *tls = tfw_tls_context(c);
+	bool hs_was_done = false;
 
-	if (TFW_FSM_TYPE(((TfwConn *)c)->proto.type) == TFW_FSM_H2)
-		tfw_h2_context_clear(tfw_h2_context(c));
+	if (TFW_CONN_PROTO((TfwConn *)c) == TFW_FSM_H2
+	    && (hs_was_done = ttls_hs_done(tls)))
```
Comment (const-t) [inline fw/tls.c:617]: Why do we need `hs_was_done`?

Comment (EvgeniiMekhanik) [reply fw/tls.c:617]: Yes we can remove this variable. First of all i plan to pass it to `tfw_cli_conn_release`, to clear &ctx->skb_head only for intialized context. But I understand, that we cant do it in a good way. Better to clear &ctx->skb_head in context cleanup function

**Pull request merged**